Revert deprecation of __sleep  and __wakeup

nicolas-grekas opened the pull request:
Without precluding the result of the vote at https://wiki.php.net/rfc/soft-deprecate-sleep-wakeup

I'm trying to help PHP 8.5 RMs get ready for RC2, as advised in this message:
https://externals.io/message/128643#128659

This PR reverts the 4 commits on the topic, related to PRs #19682 and #19435

Reviews welcome to ensure this can be merged on Oct. 4th if the vote passes.

nicolas-grekas commented:
/cc @DanielEScherzer @edorian

nicolas-grekas commented:
I'm wondering if it's worth reverting #19827 given that the affected classes already have `__[un]serialize` hooks since a few years now.

DanielEScherzer commented on NEWS:
^ that, as part of GA releasing the NEWS sections will be combined (https://github.com/php/php-src/blob/master/docs/release-process.md#preparing-for-the-initial-stable-version-php-xy0) but until then they should be separate

nicolas-grekas commented:
Thanks for the review, I removed the commit that reverted NEWS updates and also removed the revert of #19827, which can and should be kept to me.
I also added a commit to update the NEWS/UPGRADING files to mention the soft-deprecation.

TimWolla commented on UPGRADING:
```diff
@@ -401,10 +401,11 @@ PHP 8.5 UPGRADE NOTES
     __serialize() and __unserialize() magic methods should be used instead,
     or at the same time if compatibility with PHP 7 is required.
     RFC: https://wiki.php.net/rfc/deprecations_php_8_5#deprecate_the_sleep_and_wakeup_magic_methods
+    and https://wiki.php.net/rfc/soft-deprecate-sleep-wakeup
```
I believe this is the usual formatting we use:
```suggestion
    RFC: https://wiki.php.net/rfc/soft-deprecate-sleep-wakeup
```

theodorejb commented on ext/session/tests/bug53141.phpt:
```diff
@@ -16,7 +16,7 @@ class Foo
     {
         new Bar;
-        return [];
+	return [];
```
Why is the formatting changed here?

nicolas-grekas replied:
addressed while rebasing

edorian reviewed (approved):
On the, likely, condition that the RFC passes: 👍 RM wise.

Thank you for getting everything ready, making it easy to stay within the plan made to get this sorted cleanly. Much appreciated.

DanielEScherzer commented:
I was planning to review this today and then merge it, sorry for the delay - looks good to me and I guess has already been merged
